policy/policy-data_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package policy
+
+import (
+	"testing"
+
+	"github.com/letsencrypt/boulder/sa"
+	"github.com/letsencrypt/boulder/test"
+)
+
+var (
+	rA = DomainRule{
+		Host: "a.com",
+		Type: whitelisted,
+	}
+	rB = DomainRule{
+		Host: "b.com",
+		Type: blacklisted,
+	}
+	rC = DomainRule{
+		Host: "d.c.com",
+		Type: blacklisted,
+	}
+	rD = DomainRule{
+		Host: "a.d.com",
+		Type: whitelisted,
+	}
+	rE = DomainRule{
+		Host: "d.com",
+		Type: blacklisted,
+	}
+)
+
+func TestLoadAndDump(t *testing.T) {
+	p, cleanup := padbImpl(t)
+	defer cleanup()
+
+	err := p.LoadRules([]DomainRule{rA, rB, rC})
+	test.AssertNotError(t, err, "Couldn't load rules")
+
+	r, err := p.DumpRules()
+	test.AssertNotError(t, err, "Couldn't dump rules")
+
+	test.AssertEquals(t, len(r), 3)
+}
+
+func TestGet(t *testing.T) {
```

Comment:
Corresponding to the method split above, you need two tests here:
* TestWhitelist: Test that only exact matches count as whitelisted
* TestBlacklist:
  * Test that exact matches are blacklisted
  * Test that subdomains are blacklisted
  * Test that substrings are *not* blacklisted (e.g., "mail.com" if "mailthing.com" is on the blacklist
  
You also need to add a test to the overall PA (once the WL/BL logic moves up there) to test that the blacklist overrides the whitelist.